### PR TITLE
chore(java): handling of fallback strategy

### PR DESCRIPTION
**Pull request by gastonfournier**
## About the changes
After doing a first integration in Java SDK these are feature that were missing in the Engine

### CODE REVIEW

File: java-engine/src/main/java/io/getunleash/engine/CustomStrategiesEvaluator.java [hunk 1]
```diff
@@ -65,14 +73,20 @@ private List<MappedStrategy> getFeatureStrategies(FeatureDefinition feature) {
             mappedStrategies.add(
                     new MappedStrategy("customStrategy" + (index++), impl, strategyDefinition));
         }
+        if (fallbackStrategy != null) {
+            mappedStrategies.add(
+                    new MappedStrategy(
+                            "customStrategy" + index,
+                            fallbackStrategy,
+                            new StrategyDefinition("fallback", Collections.emptyMap())));
```
Comment (gastonfournier) [inline java-engine/src/main/java/io/getunleash/engine/CustomStrategiesEvaluator.java:81]: The name doesn't really matter, this strategy has to be available for all feature toggles

File: java-engine/src/test/java/io/getunleash/engine/UnleashEngineTest.java [hunk 2]
```diff
@@ -100,6 +101,7 @@ void testGetVariant() throws Exception {
     @Test
     public void testClientSpec() throws Exception {
         ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (gastonfournier) [inline java-engine/src/test/java/io/getunleash/engine/UnleashEngineTest.java:104]: Just for easier evolution: don't fail when new fields are added.

**Comment by sighphyre** [comment]
I'm not convinced this is the correct thing to do here. This looks like a very odd feature in the Java SDK. Can we move towards deprecating that rather?

**Comment by gastonfournier** [comment]
> I'm not convinced this is the correct thing to do here. This looks like a very odd feature in the Java SDK. Can we move towards deprecating that rather?

I can mark it as deprecated in the engine's public interface and then in the SDK, but right now this is needed for backward compatibility

**Review by sighphyre — APPROVED** [review]
G

**Pull request merged**